### PR TITLE
Iox #381 Switch from message queue to UNIX domain sockets

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer
* Switch from message queue communication to UNIX domain sockets
* Disable some implementation specifc tests that will be refactored as part of #381 in next PR
* Renaming all message queue wording to IPC channel

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Closes partly #381

### CODE REVIEW

File: iceoryx_posh/test/moduletests/test_mq_interface.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ TEST_F(CMqInterface_test, MqBase_StringCTor)
     CMqInterface_StringCTor<MqBase>();
 }
 
-TEST_F(CMqInterface_test, MqBase_Receive)
+TEST_F(CMqInterface_test, DISABLED_MqBase_Receive)
```
Comment (elBoberido) [inline iceoryx_posh/test/moduletests/test_mq_interface.cpp:241]: it seems the sins of our past are catching us again

File: iceoryx_posh/test/integrationtests/test_mq_interface_startup_race.cpp [hunk 2]
```diff
@@ -106,13 +106,13 @@ class CMqInterfaceStartupRace_test : public Test
...
     IpcChannelType::result_t m_roudiQueue{
-        IpcChannelType::create(roudi::MQ_ROUDI_NAME, IpcChannelMode::BLOCKING, IpcChannelSide::SERVER)};
+        IpcChannelType::create(roudi::IPC_CHANNEL_ROUDI_NAME, IpcChannelMode::BLOCKING, IpcChannelSide::SERVER)};
     std::mutex m_appQueueMutex;
     IpcChannelType::result_t m_appQueue;
 };
 
 #if !defined(__APPLE__)
-TEST_F(CMqInterfaceStartupRace_test, ObsoleteRouDiMq)
+TEST_F(CMqInterfaceStartupRace_test, DISABLED_ObsoleteRouDiMq)
```
Comment (elBoberido) [inline iceoryx_posh/test/integrationtests/test_mq_interface_startup_race.cpp:115]: we have to check which of these make sense after the startup procedures is redone

**Review by elBoberido — APPROVED** [review]

File: iceoryx_posh/test/moduletests/test_mq_interface.cpp (hunk 1 above)
Comment (dkroenke) [reply iceoryx_posh/test/moduletests/test_mq_interface.cpp:241]: @mossmaurice please add a checkbox to the issue that we need to handle these tests (Either remove or rewrite).

Comment (mossmaurice) [reply iceoryx_posh/test/moduletests/test_mq_interface.cpp:241]: The checkbox was already added in #381

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp [hunk 3]
```diff
@@ -115,20 +115,20 @@ class RouDi
...
                          const version::VersionInfo& versionInfo);
 
-    /// @brief Creates a unique ID which can be used to check outdated mqueue transmissions
+    /// @brief Creates a unique ID which can be used to check outdated IPC channel transmissions
     /// @return a unique, monotonic and consecutive increasing number
     static uint64_t getUniqueSessionIdForProcess();
 
   private:
-    void mqThread();
+    void ipcChannelThread();
```
Comment (dkroenke) [inline iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:123]: We should use the chance to clear up the naming a bit. ipcChannelThread does not really reflect what's happenning in there. Basically we are retrieving ipc channel messages from the runtime and forward them to `processMessage`.
Possible name:
`retrieveAndProcessRuntimeMessageThread`

Comment (elBoberido) [reply iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:123]: in my previous life, we called this stuff `CmdHandler` ... maybe I'm getting old :D

Comment (mossmaurice) [reply iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:123]: Good point, renamed `processThread()` accordingly.

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp [hunk 4]
```diff
@@ -141,7 +141,7 @@ class RouDi
 
   private:
     std::thread m_processManagementThread;
-    std::thread m_processMQThread;
+    std::thread m_processRuntimeMessagesThread;
```
Comment (elBoberido) [inline iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:144]: This is confusing. Why does something called `IpcThreadStart` control something called `m_processRuntimeMessagesThread`?

Comment (mossmaurice) [reply iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:144]: Ups :blush: Missed the `IpcThreadStart`. Fixed.

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp [hunk 5]
```diff
@@ -41,9 +41,9 @@ using namespace iox::units::duration_literals;
 class RouDi
 {
   public:
-    // indicate whether the message queue thread will start directly or deferred
+    // indicate whether the IPC channel thread will start directly or deferred
     // this is important for derived classes which may need to initialize their members before the thread starts
-    enum class MQThreadStart
+    enum class RuntimeMessagesThreadStart
```
Comment (elBoberido) [inline iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:46]: the documentation talks about `IPC channel threads` ... oh, and please make it `/// @brief` :D

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp [hunk 6]
```diff
@@ -82,9 +82,9 @@ class RouDi
     virtual ~RouDi();
 
   protected:
-    /// @brief Starts the roudi message queue thread
+    /// @brief Starts the roudi IPC channel thread
     /// Once this is done, applications can register and Roudi is fully operational.
-    void startMQThread();
+    void startProcessRuntimeMessagesThread();
```
Comment (elBoberido) [inline iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:87]: same here

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp (hunk 5 above)
Comment (mossmaurice) [reply iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:46]: :laughing: Fixed

File: iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp (hunk 6 above)
Comment (mossmaurice) [reply iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp:87]: Adapted

**Review by elBoberido — DISMISSED** [review]

**Review by elBoberido — DISMISSED** [review]

**Review by elBoberido — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Pull request merged**